2016/proposals/rai-vivek.md

```diff
@@ -0,0 +1,263 @@
+lays out following objectives:
+
+* implement a server to store the results of installation-testing scripts (i.e.
+platform people use, which software dependency worked and which didn't, error
+logs)
+* enhance the installation-test scripts by adding an option for users to collect results and send
+them to the server
+
+## Technical Details
+
+The technical details of the project can be organized into following:
+
+* Collectable information - machine information, report
+* Backend - design of RESTful API, framework, database, schema
+* Frontend - import data, HTML reports, visualizations
+* Upgrading installation-test scripts
+* Documentation and integration tests
+
+### Collectable information
+
+In this section, we discuss the parameters that can be collected by the SwC’s
+installation-scripts to help us analyze and improve installation setup for future
+workshops.
+
+The goal is to collect - anonymized, non-identifiable machine information and
+status report of the installation-script.
+
+#### Machine information
+
+* Platform (or System)
+* Architecture (or Machine)
+* Node, Processor
+* OS (or Distribution), Release version, Name
+
+Most of this information can be collected using the `platform` Python module.
+A snippet of useful commands is shown below.
+
+```
+import platform 
+platform.uname() # cross platform
+platform.win32_ver() # windows
+platform.mac_ver() # mac
+platform.linux_distribution() # linux
+```
+
+#### Status report
+
+* Software check status (pass/fail) along with relevant messages
+* Version number of the found (installed) softwares
+* Any error that may have occurred in the script itself
+* Process information
+* Timestamp
+* (Optional) User email
+
+### Server
+
+The server will collect the diagnostic report submitted by installation-scripts
+when they are run on the user machine.
+
+A simple design of the server is shown below:
+
+#### API
+
+The following steps are involved in the process:
+
+* Installation-script sends a `POST` request to the server endpoint (for example, https://installation.software-carpentry.org/report).
+  > Using Flask web-development framework for Python. A robust, flexible
+  > choice that is easy to develop and maintain. Scalability is not an issue for
+  > our use case.
+
+* Server verifies the data, open database connection, updates the record.
+  > As suggested in the ideas sections, SQLite will be a low-barrier, easier to
+  > maintain database, appropriate for our needs.
+  > TODO: Investigate the worst case scenario of using SQLite.
+  > TODO: Database schema
+
+* Send a status response back to the client. The response can contain feedback,
+  suggestions or perhaps link to some survey.
+
+* Ability to export diagnostic data into JSON or CSV format.
+  > It’ll allow us to separate the user interface from backend creating a modular
+  > setup.
+
+### User interface
+
+* Authorization though a simple login system (Optional)
```

Comment:
We can drop this. We want to have a low barrier for user to send us the information.